etl/steps/data/garden/state_capacity/2023-11-10/information_capacity_dataset.py

```diff
@@ -0,0 +1,167 @@
+"""Load a meadow dataset and create a garden dataset."""
+
+import owid.catalog.processing as pr
+from owid.catalog import Dataset, Table
+
+from etl.data_helpers import geo
+from etl.helpers import PathFinder, create_dataset
+
+# Get paths and naming conventions for current step.
+paths = PathFinder(__file__)
+
+
+def run(dest_dir: str) -> None:
+    #
+    # Load inputs.
+    #
+    # Load meadow dataset.
+    ds_meadow = paths.load_dataset("information_capacity_dataset")
+
+    # Read table from meadow dataset.
+    tb = ds_meadow["information_capacity_dataset"].reset_index()
+
+    # Load population data.
+    ds_pop = paths.load_dataset("population")
+
+    # Load regional data
+    ds_regions = paths.load_dataset("regions")
+    tb_country_list = ds_regions["regions"].reset_index()
+    tb_country_list = tb_country_list[
+        (tb_country_list["region_type"] == "country") & ~(tb_country_list["is_historical"])
+    ].reset_index(drop=True)
+
+    #
+    # Process data.
+    #
+
+    # Drop country id columns
+    tb = tb.drop(columns=["ccodecow", "vdemcode"])
+
+    # Harmonize country names.
+    tb = geo.harmonize_countries(
+        df=tb, countries_file=paths.country_mapping_path, excluded_countries_file=paths.excluded_countries_path
+    )
+
+    # Reset index after dropping excluded countries
+    tb = tb.reset_index(drop=True)
+
+    # Add regional aggregations
+    tb = regional_aggregations(tb, ds_pop, tb_country_list)
+
+    tb = tb.set_index(["country", "year"], verify_integrity=True)
+
+    #
+    # Save outputs.
+    #
+    # Create a new garden dataset with the same metadata as the meadow dataset.
+    ds_garden = create_dataset(
+        dest_dir, tables=[tb], check_variables_metadata=True, default_metadata=ds_meadow.metadata
+    )
+
+    # Save changes in the new garden dataset.
+    ds_garden.save()
+
+
+def regional_aggregations(tb: Table, ds_pop: Dataset, tb_country_list: Table) -> Table:
```

Comment:
Isn't rolling included for Tables?
I have this error:
```
  File "/Users/parriagadap/etl/etl/steps/data/garden/state_capacity/2023-11-10/information_capacity_dataset.py", line 95, in add_new_indicators
    tb["census_10_years"] = tb.groupby("country")["census"].rolling(10, min_periods=1).sum()
                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/parriagadap/etl/lib/catalog/owid/catalog/tables.py", line 997, in func
    raise NotImplementedError()
NotImplementedError
```

What would the alternative be here, @Marigold?

Reply:
I'll check now if it's trivial to implement `rolling`, but for now, you can do the following (a big cumbersome) workaround:
```
Variable(pd.DataFrame(tb).groupby("country")["census"].rolling(10, min_periods=1).sum()).copy_metadata(tb["census"])
```
Does that make sense? (I couldn't test it because I couldn't find it anywhere in the code in your current branch).

Reply:
I've realised it's not trivial to implement (we may need to define a special `Rolling` object). So I'll leave it for now. Feel free to create an issue if you think this can be useful.

Reply:
I think maybe what you can do is:
```
tb["census_10_years"] = pd.DataFrame(tb).groupby("country", as_index=False)["census"].rolling(10, min_periods=1).sum()["census"]
tb["census_10_years"] = tb["census_10_years"].copy_metadata(tb["census"])
```
But I haven't looked into the code in depth, so I don't know what the intention is.

Reply:
Bah, sorry, now it's in the branch. I tried that and I had `TypeError: incompatible index of inserted column with frame index`. I included `.values` to each operation and now I have `ValueError: variable must be named to have metadata` when I apply `.copy_metadata(tb["census"])`

Reply:
Tried the second option and now I have lost the short name:
```
(etl-py3.11) parriagadap@Pablos-MacBook-Pro etl % ENV=.env.parriagadap etl information_capacity --grapher
--- Detecting which steps need rebuilding...
OK (1.4s)                                                             
--- Running 3 steps:
--- 1. data://garden/state_capacity/2023-11-10/information_capacity_dataset...
2023-11-22 14:47:30,681 - [bugsnag] WARNING - No API key configured, couldn't notify
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/Users/parriagadap/etl/.venv/lib/python3.11/site-packages/click/core.py", line 1157, in __call__
    return self.main(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/parriagadap/etl/.venv/lib/python3.11/site-packages/click/core.py", line 1078, in main
    rv = self.invoke(ctx)
         ^^^^^^^^^^^^^^^^
  File "/Users/parriagadap/etl/.venv/lib/python3.11/site-packages/click/core.py", line 1434, in invoke
    return ctx.invoke(self.callback, **ctx.params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/parriagadap/etl/.venv/lib/python3.11/site-packages/click/core.py", line 783, in invoke
    return __callback(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/parriagadap/etl/etl/run_python_step.py", line 34, in main
    _import_and_run(path, dest_dir)
  File "/Users/parriagadap/etl/etl/run_python_step.py", line 54, in _import_and_run
    module.run(dest_dir)
  File "/Users/parriagadap/etl/etl/steps/data/garden/state_capacity/2023-11-10/information_capacity_dataset.py", line 61, in run
    ds_garden = create_dataset(
                ^^^^^^^^^^^^^^^
  File "/Users/parriagadap/etl/etl/helpers.py", line 164, in create_dataset
    ds.add(table, formats=formats)
  File "/Users/parriagadap/etl/lib/catalog/owid/catalog/datasets.py", line 143, in add
    table_filename = join(self.path, table.metadata.checked_name + f".{format}")
                                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/parriagadap/etl/lib/catalog/owid/catalog/meta.py", line 444, in checked_name
    raise Exception("table has no short_name")
Exception: table has no short_name

COMMAND: poetry run run_python_step data://garden/state_capacity/2023-11-10/information_capacity_dataset /Users/parriagadap/etl/data/garden/state_capacity/2023-11-10/information_capacity_dataset
```

Reply:
That's a different problem. You lose the table short name when you merge the main table with another that doesn't have a short name (line 90, inside `add_new_indicators`). To fix it:
```
    # Merge this table with tb
    tb = pr.merge(tb_countries_years, tb, how="left", on=["country", "year"], short_name=tb.metadata.short_name)

```

Reply:
Thanks! I won't forget to add it for merges/concats